File: CODIGO 10/work.cs
using System;

public class CloudAltitudeMeasurement
{
    public static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());

        int[] heights = new int[n];
        for (int i = 0; i < n; i++)
        {
            string[] data = Console.ReadLine().Split(" ");

            int d1 = int.Parse(data[0]);
            float a = float.Parse(data[1]);
            float b = float.Parse(data[2]);

            heights[i] = (int)Math.Round(tan(b) * d1 / (tan(a) - tan(b)));
        }

        foreach (int height in heights)
        {
            Console.WriteLine(height);
        }
    }
}

File: CODIGO 10/work.lua
function cloud_altitude(d1, a, b)
    h = math.tan(b) * d1 / (math.tan(a) - math.tan(b))
    return math.floor(h)
  end
  
  n = tonumber(io.read())
  
  heights = {}
  for i = 1, n do
    d1, a, b = io.read():split()
    heights[i] = cloud_altitude(tonumber(d1), tonumber(a), tonumber(b))
  end
  
  for i, h in ipairs(heights) do
    print(h)
  end
  